### ISSUE TITLE
Block websites during focus sessions

### ISSUE THREAD

**Issue by schmich** [schmich/marinara#32]
Users should be able to block themselves from accessing specified websites during focus sessions.

From Justin on Chrome store:
> Missing an important feature: blocking some websites while on a work stretch. Other tools have this. Uninstalled this.

Implementation notes:
- New section in settings to specify websites in a list or textarea
- Domain wildcards? Only use apex domains?
- Support comma/newline/tab/semicolon-separated lists
- Sync or local storage? Sync storage limits capacity
- What if focus session is paused?
- Create custom page to notify user when they try to access excluded websites
- Link to extension settings from blocked page

**Comment by freetitelu** [comment]
There are too many nuances. And in the end I removed Pomodoro-app and still looked for the cats and boobs. It is a delusion that some chrome-app can save from procrastination. I think your Pomodoro is so perfect for your tasks - just remind when it's time to rest. PS: People who want "a blocking some websites", just need to spend time adjusting it - for another 5 minutes procrastination. I know all about this (perhaps this feature is only for the coverage of this contingent of users. And here is only one way - to turn off the Internet within a radius of several thousand kilometers, so that the rest of the indignant users come and beat you for violation of the block). And sorry for my english.

**Comment by schmich** [comment]
I agree with you, which is why I'm torn on this feature. Personally, I'd rather train and build willpower/discipline/self-control/determination instead of relying on a browser extension. On the other hand, people are imperfect, and maybe this helps them just enough to build better habits. It's hard to say.

Francesco Cirillo, the creator of the Pomodoro Technique, [encourages a physical approach to his technique](https://en.wikipedia.org/wiki/Pomodoro_Technique#Tools), so I can imagine he would disapprove of using software at all. Yet here we are, and I know this extension helps a lot of people be productive day-to-day.

I have my opinion on blocking websites, and I'd rather not bloat the extension, but ultimately, I listen to users, and if the demand is strong enough, I will add it. For now, though, it is in the "Future" bucket.

Thanks for the feedback.

**Comment by Hongbo-Miao** [comment]
Some block functions like this Focus Mode might be helpful.
https://chrome.google.com/webstore/detail/focus-mode/jhfefcgbjemmakkkaabcdofehlegadbl
